ClimbingGearBackend.Tests/Controllers/GearControllerTest.cs

```diff
@@ -0,0 +1,208 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using ClimbingGearBackend.Controllers;
+using ClimbingGearBackend.Models;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+
+namespace ClimbingGearBackend.Tests
+{
+  public abstract class GearControllerTest
+  {
+    protected GearControllerTest(DbContextOptions<ClimbingGearContext> contextOptions)
+    {
+      ContextOptions = contextOptions;
+
+      Seed();
+    }
+    protected DbContextOptions<ClimbingGearContext> ContextOptions { get; }
+
+    private void Seed()
+    {
+      using (var context = new ClimbingGearContext(ContextOptions))
+      {
+        context.Database.EnsureDeleted();
+        context.Database.EnsureCreated();
+
+        var one = new Gear {
+          Id = 1,
+          Name = "Cam",
+          Description = "Just a cam",
+          Brand = "TestCams" };
+        
+        var two = new Gear {
+          Id = 2,
+          Name = "Nut",
+          Description = "Just a nut",
+          Brand = "TestNuts" };
+        
+        var three = new Gear {
+          Id = 3,
+          Name = "Carabiner",
+          Description = "Just a carabiner",
+          Brand = "TestCarabiners" };
+
+        context.AddRange(one, two, three);
+
+        context.SaveChanges();
+      }
+    }
+    [Fact]
+    public async Task GetGear_CallGetGearWithNoId_ShouldReturnAllGear()
+    {
+      using (var context = new ClimbingGearContext(ContextOptions))
+      {
+        var controller = new GearController(context);
+
+        var result = await controller.GetGear();
+        
+        var gearList = new List<GearDTO>(result.Value);
+        Assert.Equal(3, gearList.Count);
+        Assert.Equal("Cam", gearList[0].Name);
+        Assert.Equal("Nut", gearList[1].Name);
+        Assert.Equal("Carabiner", gearList[2].Name);
+      }
+    }
+
+    [Fact]
+    public async Task GetGear_CallWithAnIdThatExists_ShouldReturnCorrectGear()
+    {
+      using (var context = new ClimbingGearContext(ContextOptions))
+      {
+        var controller = new GearController(context);
+
+        var result = await controller.GetGear(1);
+
+        var gear = result.Value;
+        Assert.Equal("Cam", gear.Name);
+      }
+    }
+
+    [Fact]
+    public async Task GetGear_CallWithIdThatDoesntExist_ShouldReturnNotFound()
+    {
+      using (var context = new ClimbingGearContext(ContextOptions))
+      {
+        var controller = new GearController(context);
+
+        var result = await controller.GetGear(42);
+
+        var resultObject = new NotFoundObjectResult(result.Value);
```

Comment:
I don't think this is what you want to do for this test. You don't want to construct your own `NotFound` result, you want to make sure the result returned from `GetGear` was a not found result.

From a little research, I think you want something like this:

```cs
var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
```

This goes for the `as XResult` expressions in the tests below.